node-ui/src/components/starknet/StarknetWallet.tsx

```diff
@@ -0,0 +1,225 @@
+        padding-left: 0.5rem;
+      }
+    }
+    .logout-wrap {
+      padding-top: 1rem;
+      font-size: 14px;
+      color: #FFFFFF;
+      text-align: center;
+      cursor: pointer;
+    }
+
+    .back-wrap {
+      padding-top: 1rem;
+      font-size: 14px;
+      color: #FFFFFF;
+      text-align: center;
+      cursor: pointer;
+    }
+
+    .error-message {
+      color: red;
+      font-size: 14px;
+      font-weight: 500;
+      margin-top: 0.5rem;
+    }
+  }
+`;
+
+interface LoginWithStarknetProps {
+  navigateBack: () => void;
+  isLogin: boolean;
+  ready: boolean;
+  walletLogin: (type: string, setErrorMessage: (msg: string) => void) => void;
+  starknetInstance: StarknetWindowObject | null;
+  argentXId: string;
+  walletSignatureData: WalletSignatureData | null;
+  signMessage: (setErrorMessage: (msg: string) => void) => void;
+  errorMessage: string;
+  setErrorMessage: (msg: string) => void;
+  logout: (setErrorMessage: (msg: string) => void) => void;
+  changeMetamaskNetwork: (network: string, setErrorMessage: (msg: string) => void) => void;
+}
+
+export function StarknetWallet({
+  navigateBack,
+  isLogin,
+  ready,
+  walletLogin,
+  starknetInstance,
+  argentXId,
+  walletSignatureData,
+  signMessage,
+  errorMessage,
+  setErrorMessage,
+  logout,
+  changeMetamaskNetwork,
+}: LoginWithStarknetProps) {
+  const t = translations.loginPage.starknetLogin;
+
+  if (!ready) {
+    return <Loading />;
+  }
+
+  return (
+    <Fragment>
+      <Wrapper>
+        <div className="wrap">
+          <span className="title">{isLogin ? t.title : t.titleRootKey}</span>
+          <div className="subtitle-wrap">
+            <span className="subtitle-content">{t.subtitle}</span>
+          </div>
+          {!starknetInstance && (
+            <header className="wallet-options-wrap">
+              <span className="wallet-btn" onClick={() => walletLogin('argentX', setErrorMessage)}>
+                {t.loginWithArgentX}
+              </span>
+              <span className="wallet-btn" onClick={() => walletLogin('metamask', setErrorMessage)}>
+                {t.loginWithMetaMask}
+              </span>
+            </header>
+          )}
+          {starknetInstance && walletSignatureData && (
+          <>
+            {starknetInstance?.id !== argentXId && (
+              <div className="metamask-network">
+                <label htmlFor="network">{t.currentNetwork}:</label>
+                <select name="network" defaultValue={ starknetInstance.chainId === constants.StarknetChainId.SN_MAIN ? constants.NetworkName.SN_MAIN : constants.NetworkName.SN_SEPOLIA } onChange={(e) => changeMetamaskNetwork(e.target.value, setErrorMessage)}>
+                  <option value={constants.NetworkName.SN_MAIN}>Mainnet</option>
+                  <option value={constants.NetworkName.SN_SEPOLIA}>Sepolia</option>
+                </select>
+              </div>
+            )}  
+            <div className="sign-wrap">
```

Comment:
I can do this where it is appropriate, but there is a difference between wallets/protocols in their layouts and integrations. Also, since this is written with styled components they are separated styles so they don't affect each other.